### ISSUE TITLE
Moving dropq code into Tax-Calculator

### ISSUE THREAD

**Issue by talumbau** [open-source-economics/Tax-Calculator#859]
We've had issues keeping `dropq` and `taxcalc` in sync recently. Martin made the suggestion recently in this comment in the drop-q repo:

https://github.com/open-source-economics/drop-q/issues/4#issuecomment-233747542

that we might be better off moving the dropq code assets into Tax-Calculator. Now that the drop-q repo is publicly available, I am inclined to agree. The benefits would be:
- changes to Tax-Calculator would stay synchronized with changes to dropq, because there would be a single test suite for both
- deployment of TaxBrain would be easier because there would be one fewer repositories to version, deploy, and coordinate
- more eyes would be on the dropq repo, which undoubtedly could use additional refactoring

The downside that I would see would be additional complexity living in Tax-Calculator with code assets that some might not need at all. I think that people uninterested in the dropq code could just ignore it though. My suggestion would be to put it in a subpackage called `dropq`. Thus, all dropq functionality would be available in `taxcalc.dropq.*`.

**Comment by martinholmer** [comment]
@talumbau said after mentioning several advantages to moving drop-q logic into Tax-Calculator:
> The downside that I would see would be additional complexity living in Tax-Calculator with code assets that some might not need at all.  I think that people uninterested in the dropq code could just ignore it though.

There is plenty already in Tax-Calculator that most new users will have no interest in, so I agree that a little more to ignore would not be an issue.

@talumbau then said:
> My suggestion would be to put it [drop-q logic] in a subpackage called dropq.  Thus, all dropq functionality would be available in taxcalc.dropq.*.

Let me see if I understand your suggestion?
There would be a new tax-calculator/dropq directory, right?
And that new directory would contain just the following, right?
```
__init__.py
_version.py
dropq.py
utils.py
tests
   test_dropq.py
   test_utils.py
```
If I'm confused, please set me straight.

@talumbau @MattHJensen @feenberg @Amy-Xu @GoFroggyRun @zrisher @codykallen

**Comment by martinholmer** [comment]
Issue #859 is in the process of being resolved by pull request #1015.

**Comment by zrisher** [comment]
The above sounds good to me, except that the tests go in `taxcalc/tests` (or even `taxcalc/tests/dropq`) instead of `dropq/tests`. The PR is already doing this correctly. That appears to be the commonly used file structure.

**Comment by martinholmer** [comment]
@zrisher said:
> The above [directory layout] sounds good to me, except that the tests go in taxcalc/tests (or even taxcalc/tests/dropq) instead of dropq/tests. The PR is already doing this correctly. That appears to be the commonly used file structure.

Yes, you are right.  And, as you point out, pull request #1015 is putting the unit tests for the dropq code in the `taxcalc/tests/test_dropq.py` file.

**Comment by martinholmer** [comment]
Pull request #1015, which was merged into the master branch in early November 2016, resolves issue #859.
